src/api/services/AddressService.ts

```diff
@@ -0,0 +1,108 @@
+import * as Bookshelf from 'bookshelf';
+import { inject, named } from 'inversify';
+import { Logger as LoggerType } from '../../core/Logger';
+import { Types, Core, Targets } from '../../constants';
+import { validate, request } from '../../core/api/Validate';
+import { NotFoundException } from '../exceptions/NotFoundException';
+import { AddressRepository } from '../repositories/AddressRepository';
+import { Address } from '../models/Address';
+import { AddressCreateRequest } from '../requests/AddressCreateRequest';
+import { AddressUpdateRequest } from '../requests/AddressUpdateRequest';
+import { RpcRequest } from '../requests/RpcRequest';
+
+
+export class AddressService {
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Repository) @named(Targets.Repository.AddressRepository) public addressRepo: AddressRepository,
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
+    ) {
+        this.log = new Logger(__filename);
+    }
+
+    @validate()
+    public async rpcFindAll( @request(RpcRequest) data: any): Promise<Bookshelf.Collection<Address>> {
+        return this.findAll();
+    }
+
+    public async findAll(): Promise<Bookshelf.Collection<Address>> {
+        return this.addressRepo.findAll();
+    }
+
+    @validate()
+    public async rpcFindOne( @request(RpcRequest) data: any): Promise<Address> {
+        return this.findOne(data.params[0]);
+    }
+
+    public async findOne(id: number, withRelated: boolean = true): Promise<Address> {
+        const address = await this.addressRepo.findOne(id, withRelated);
+        if (address === null) {
+            this.log.warn(`Address with the id=${id} was not found!`);
+            throw new NotFoundException(id);
+        }
+        return address;
+    }
+
+    @validate()
+    public async rpcCreate( @request(RpcRequest) data: any): Promise<Address> {
+        return this.create(data.params);
```

Comment:
data.params is an array which you need to convert to address object, see for example escrowService.rpcCreate